Add CSF 2→3 banner to docs code snippets

kylegach opened the pull request:
Also added an "Upgrading from CSF 2 to CSF 3 section to [`/docs/api/csf`](https://storybook.js.org/docs/react/api/csf), which is largely based on https://storybook.js.org/blog/component-story-format-3-0/#new-in-csf-3-0.

See corresponding `frontpage` PR: https://github.com/storybookjs/frontpage/pull/437

**Preview:**
![Screenshot of code snippet banner pointing to upgrade guide and corresponding CSF 2 example](https://user-images.githubusercontent.com/486540/205178561-1234f185-8700-4a9f-9022-f7198947e726.png)

## How to test

To test it locally:

1. Check out the `kylegach/dx-392-build-csf-2--csf-3-snippet-banner` branch in the https://github.com/storybookjs/frontpage repo
1. In `frontpage`
	1. Run `yarn extract-monorepo-docs add-csf-2-to-3-banner`
	2. Run `GATSBY_SKIP_ADDON_PAGES=true yarn start`

That will pull down this branch's docs into your local `frontpage` and run the site locally. The local build won't have all of the redirects, so start here to avoid 404s: http://localhost:8000/docs/react/why-storybook.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

kylegach committed:
Update links in Addon knowledge base doc

- Point to `main` instead of `master`
- Use Chromatic branch permalink

kylegach committed:
Add CSF 2->3 attributes to docs CodeSnippets

kylegach committed:
Update snippets to CSF 3

- Fix some minor errors

kylegach committed:
Add CSF 2->3 upgrade docs & snippets

kylegach commented on docs/snippets/common/csf-3-example-starter.js.mdx:
```diff
@@ -0,0 +1,8 @@
+// CSF 3
+import { Button } from './Button';
+
+export default { component: Button };
```
I could not find an example for `web-components`, which as of this comment will use this `common` snippet.

Would appreciate if someone can confirm that is correct.

shilman replied:
All of the renderer-independent stories use `default.component` for `web-components`.

Example: https://github.com/storybookjs/storybook/blob/next/code/addons/actions/template/stories/basics.stories.ts#L6

And you can see that value getting set up here: https://github.com/storybookjs/storybook/blob/next/code/renderers/web-components/template/components/index.js#L9

Users wouldn't use this much indirection. Here's a more literal example (that happens not to use the default render function, but could): https://github.com/storybookjs/storybook/blob/next/code/renderers/web-components/template/stories/demo-wc-card.stories.js#L5

kylegach commented on docs/api/csf.md:
```diff
@@ -238,3 +253,154 @@ For this particular example, you could achieve the same result in different ways
 
 The first option is the recommended solution if you follow the best practice of starting story exports with an uppercase letter (i.e., use UpperCamelCase).
+
+## Upgrading from CSF 2 to CSF 3
```
I could not determine a suitable spot in the TOC, so I went with a new section on this page. But if someone thinks it deserves its own page, I'm open to suggestions on where it would be placed.

kylegach replied:
As discussed elsewhere, rather than move or duplicate the section, I added a link to it.

kylegach committed:
Address comments

- Update web-components snippet per https://github.com/storybookjs/storybook/pull/20053#discussion_r1037861514

kylegach committed:
Address comments

- Add links in upgrade section to other, reference sections

kylegach commented:
@jonniebigodes — Feel free to merge, if I adequately addressed your comment!

jonniebigodes reviewed (approved):
Thanks for addressing the feedback so promptly @kylegach. Appreciate it 🙏 ! I'm good on it. Merge when able.
